sdk/iot/pnp/src/az_iot_pnp_client_telemetry.c

```diff
@@ -59,33 +60,51 @@ AZ_NODISCARD az_result az_iot_pnp_client_telemetry_get_publish_topic(
   AZ_PRECONDITION_IS_NULL(reserved);
   AZ_PRECONDITION_NOT_NULL(out_mqtt_topic);
 
+  az_span written_span;
+  int32_t written = 0;
+
   AZ_RETURN_IF_FAILED(az_iot_hub_client_telemetry_publish_topic_get(
-      &client->_internal.iot_hub_client, NULL, mqtt_topic, &mqtt_topic));
+      &client->_internal.iot_hub_client,
+      NULL,
+      az_span_slice(mqtt_topic, written, -1),
+      &written_span));
+
+  written += az_span_size(written_span);
```

Comment:
I could use an extra eyes on this to review/verify the correctness.

Reply:
@jspaith